pkg/utils/cachedClient/get_cached_client.go

```diff
@@ -17,12 +18,14 @@ import (
 
 // GetCachedClient returns a controller runtime client with the cache initialized only for the resources added to
 // the scheme.
-func GetCachedClient(ctx context.Context, scheme *runtime.Scheme) (client.Client, error) {
-	conf := ctrl.GetConfigOrDie()
+func GetCachedClient(ctx context.Context, conf *rest.Config, scheme *runtime.Scheme) (client.Client, error) {
```

Comment:
Since config is kinda optional, I would put it as the last parameter of the function.